modal/dict.py

```diff
@@ -134,7 +137,9 @@ class _Dict(_Provider[_DictHandle]):
     ```
     """
 
-    def __init__(self, data={}):
+    @typechecked
+    @staticmethod
+    def new(data={}):
```

Comment:
Don't we use `Obj.from_x` naming for our other providers? Maybe we should do that here as well, since it takes an argument (albeit optionally, so I can see the point for `new` as well...)